library/common/stream_info/extra_stream_info.cc

```diff
@@ -6,50 +6,60 @@ namespace Envoy {
 namespace StreamInfo {
 namespace {
 
-void setFromOptional(int64_t& to_set, const absl::optional<MonotonicTime>& time) {
+void setFromOptional(int64_t& to_set, const absl::optional<MonotonicTime>& time,
+                     int64_t offset_ms) {
   if (time.has_value()) {
-    to_set = std::chrono::duration_cast<std::chrono::milliseconds>(time.value().time_since_epoch())
+    to_set = offset_ms +
+             std::chrono::duration_cast<std::chrono::milliseconds>(time.value().time_since_epoch())
                  .count();
   }
 }
 
-void setFromOptional(int64_t& to_set, absl::optional<std::chrono::nanoseconds> time, long offset) {
-  if (time.has_value()) {
-    to_set = offset + std::chrono::duration_cast<std::chrono::milliseconds>(time.value()).count();
-  }
-}
-
 } // namespace
 
 const std::string& ExtraStreamInfo::key() {
   CONSTRUCT_ON_FIRST_USE(std::string, "envoy_mobile.extra_stream_info");
 }
 
-void setFinalStreamIntel(StreamInfo& stream_info, envoy_final_stream_intel& final_intel) {
+void setFinalStreamIntel(StreamInfo& stream_info, TimeSource& time_source,
+                         envoy_final_stream_intel& final_intel) {
+  final_intel.request_start_ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+                                     stream_info.startTime().time_since_epoch())
+                                     .count();
+  int64_t offset_ms =
```

Comment:
I think it's not clear what we're doing with this offset - mind commenting?

Reply:
Added a little comment.

Done.